File: server/ruby/app/infrastructure/mappers/additional_payment_actions/amount_of_money_mapper.rb
module Infrastructure
  module Mappers
    module AdditionalPaymentActions
      module AmountOfMoneyMapper
        def self.map(dto)
          return nil if dto.amount.nil? || dto.currency.nil?

          amount_of_money = OnlinePayments::SDK::Domain::AmountOfMoney.new
          amount_of_money.amount = dto.amount
          amount_of_money.currency_code = dto.currency.is_a?(String) ? dto.currency : dto.currency&.value

          amount_of_money
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/additional_payment_actions/amount_of_money_mapper.py
from onlinepayments.sdk.domain.amount_of_money import AmountOfMoney
from app.application.dtos.additional_payment_actions.request_dto import AdditionalPaymentActionsRequestDto

class AmountOfMoneyMapper:

    @staticmethod
    def to_sdk_amount_of_money(dto: AdditionalPaymentActionsRequestDto):
        if dto.amount is None or dto.currency is None:
            return None

        amount_of_money = AmountOfMoney()
        amount_of_money.amount = dto.amount
        amount_of_money.currency_code = dto.currency if isinstance(dto.currency, str) else dto.currency.value

        return amount_of_money